kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java

```diff
@@ -261,12 +283,10 @@ protected void onMessage(String message) {
     try {
       WatchEvent event = readWatchEvent(message);
       Object object = event.getObject();
-      if (object instanceof HasMetadata) {
-        @SuppressWarnings("unchecked")
-        T obj = (T) object;
-        updateResourceVersion(obj.getMetadata().getResourceVersion());
-        Action action = Action.valueOf(event.getType());
-        eventReceived(action, obj);
+      if (object instanceof Status) {
```

Comment:
This is just a little bit of reordering:
 - deserializing the specific type is "more likely" to end up with something that `extends HasMetadata`
 - `Status` will be checked first